### ISSUE TITLE
Hint "CDI artifact is found but there is no beans.xml file" can't be disabled, although beans.xml is optional CDI 1.1+

### ISSUE THREAD

**Issue by m1kryn** [apache/netbeans#6399]
### Apache NetBeans version

Apache NetBeans 19 release candidate

### What happened

 I have a Gradle web application project that runs in Payara with the following problem.

 All of my `@ViewScoped` & `@Named` classes (e.g. controllers) are showing what looks like a red compile error on the class name that says "CDI artifact is found but there is no beans.xml file." However, there is a beans.xml file located in Web Pages->WEB-INF. This error does not appear in NB 18. Also, I can build and deploy the project without problems. So, apparently more of a "cosmetic" issue?

### How to reproduce

Create a Gradle Jakarta EE 10 web application that runs on Payara. Create a ViewScoped and Named backing bean within the project. Add a beans.xml file to WEB-INF.

### Did this work correctly in an earlier version?

Apache NetBeans 18

### Operating System

macOS Ventura 13.4.1

### JDK

OpenJDK 64-Bit Server VM Temurin-11.0.20+8

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

This occurs every time I use NB 19.

### Are you willing to submit a pull request?

Yes

**Comment by juneau001** [comment]
Thanks for putting the issue in the tracker.  Sorry to hear of the issues.  It looks like this is a Gradle web application utilizing Jakarta EE 10 which is encountering the issues.  Have you tried to see if you have the same issues when creating a Maven web project using Jakarta EE 10?  If you see the same issues with a Maven web project, then please make sure you try to use a new userdir to see if that helps.

**Comment by gabrielcampos** [comment]
I can confirm that this also occurs with a Maven project.

**Comment by matthiasblaesing** [comment]
I just created two fresh "Java with Maven" "Web Application" projects. One with target Jakarta EE 10 and and one with Jakarta EE 8. In both cases the project is created with a pregenerated `beans.xml` and in both cases I added a bean "TestBean". In the end I look at both of these files and get no errors:

**beans.xml (Jakarta)**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/beans_4_0.xsd"
       bean-discovery-mode="all">
</beans>
```

**TestBean.java (Jakarta)**
```java
package eu.doppelhelix.test.mavenproject2;

import java.io.Serializable;
import javax.faces.view.ViewScoped;
import javax.inject.Named;

@ViewScoped
@Named
public class TestBean implements Serializable {

}
```

**beans.xml (JavaEE)**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_2_0.xsd"
       bean-discovery-mode="all">
</beans>
```

**TestBean.java (JavaEE)**
```java
import java.io.Serializable;
import javax.faces.view.ViewScoped;
import javax.inject.Named;

@ViewScoped
@Named
public class TestBean implements Serializable {
}
```

So could you please provide a sample project, that reproduces the behavior?

**Comment by hink084** [comment]
I can't reproduce this issue if the `beans.xml` exists - in a web application it must exist in the `WEB-INF` directory and in an EJB module or JAR file it must exist in the `META-INF` directory.  However, the `beans.xml` file is **not required** for CDI v1.1+, and an error is incorrectly shown if the `beans.xml` file is not present.  This was not the same behavior in NetBeans <= v18.

<img width="366" alt="error-6399" src="https://github.com/apache/netbeans/assets/5665720/412df0a6-e70a-4a4d-9f87-793746eaffe9">


A sample project is attached. 

- NetBeans 19
- Jakarta 10
- JDK 17
- Windows 10

**Sample Project:** [6399.zip](https://github.com/apache/netbeans/files/12795582/6399.zip)

**Comment by matthiasblaesing** [comment]
> However, the beans.xml file is not required for CDI v1.1+

Ok, that makes sense. The hint, that generates that warning needs to modified. I note however, that the specification discourages implicit bean archives, 

From https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#bean_archive:
> To ensure portability between CDI Lite and CDI Full, applications are encouraged to:
> - always add a beans.xml file to an archive which contains classes with bean defining annotations;
> - never add classes with bean defining annotations to an archive without beans.xml.

I think the real question is: Why can't this hint be deactivated? If that would be the case, I would say: "Lets keep it, as it is a valid warning, for teams deciding not wanting to be bothered by this, they can deactivate the hint".

**Comment by hink084** [comment]
@matthiasblaesing I wouldn't say the specification discourages implicit bean archives.  Explicit archives are not permitted with CDI Lite.  The new default value for the `bean-discovery-mode` attribute in CDI 4.0 is changed from `all `to `annotated`.  `annotated` is also used when an empty `beans.xml` file is discovered during deployment.  This would imply implicit bean archives are now the expected default.

From CDI Full: [https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#bean_archive_full](https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#bean_archive_full)

> An explicit bean archive is an archive which contains a beans.xml file with bean-discovery-mode of all.
> 
> An implicit bean archive is:
> 
> - an archive which contains a beans.xml file that is empty, or,
> 
> - any other archive which contains one or more bean classes with a bean defining annotation as defined in [Bean defining annotations](https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#bean_defining_annotations).

To my understanding with CDI 4.0, an empty `beans.xml`, a non-existent `beans.xml`, and a `beans.xml` file with the following content are equivalent.


```XML
<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/beans_4_0.xsd"
       bean-discovery-mode="annotated"
       version="4.0">
</beans>
```

I think the section you referenced from the specification is only referring to portability between CDI Lite and CDI Full implementations to prevent unexpected deployment issues.  

> Implementations that do not support CDI Full are required to detect presence of an archive which contains a beans.xml file that has bean-discovery-mode attribute set to all and treat it as a deployment problem.

I know CDI Lite and CDI Full are new in CDI 4.0 - previously there was only the CDI Core.  Jakarta EE application containers are required to implement CDI Full.

So I'm not so sure a non-existent beans.xml is an issue here.  However, if you want to keep the NetBeans hint I would suggest changing it from the red error indicator <img width="17" alt="6399-error" src="https://github.com/apache/netbeans/assets/5665720/de4894df-2665-42e4-9d1c-ae99f70ec298"> to the yellow warning indicator <img width="17" alt="6399-warning" src="https://github.com/apache/netbeans/assets/5665720/12b71053-3c3f-46c8-8fc2-53e7bc3396e7">.

**Comment by m1kryn** [comment]
I can confirm that with NB 19 and 20-rc4, if I do as @hink084 outlines above and make sure that I also have a beans.xml file in META-INF, I no longer see the error messages. Both beans.xml files are blank.

Thanks so much to everyone on this thread for looking into this!

**Comment by bob-tee** [comment]
I came across this today using Netbeans 19.

I have a Web App that I deploy into Tomcat. I'm using the Weld DI library. I have a beans.xml file located in `src/main/webapp/WEB-INF/beans.xml`. The application builds and runs as expected. 

I get warnings in the Netbeans UI on any injected beans saying "CDI artifact is found but there is no beans.xml file.". It offers to create a beans.xml file in `src/main/resources/META-INF`. If I create this file then the warnings are gone and the application still runs as expected. I'd really prefer not to have to create a second beans.xml file. I've found it can be an empty file and Netbeans is still satisfied even if my main beans.xml file is not empty.
